cmd/kubeadm/app/cmd/phases/kubelet.go

```diff
@@ -18,184 +18,161 @@ package phases
+		Use:     "write-config-to-disk",
+		Short:   "Writes kubelet configuration to disk, either based on the --config argument or the kubeadm-config ConfigMap.",
+		Long:    kubeletWriteConfigToDiskLongDesc,
+		Example: kubeletWriteConfigToDiskExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			cfg := &kubeadmapiv1alpha2.MasterConfiguration{
-				// KubernetesVersion is not used by kubelet upload, but we set this explicitly to avoid
-				// the lookup of the version from the internet when executing ConfigFileAndDefaultsToInternalConfig
-				KubernetesVersion: "v1.9.0",
+			if len(cfgPath) == 0 {
+				kubeadmutil.CheckErr(fmt.Errorf("The --config argument is required"))
 			}
-			kubeadmscheme.Scheme.Default(cfg)
 
-			// This call returns the ready-to-use configuration based on the configuration file that might or might not exist and the default cfg populated by flags
-			internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, cfg)
+			kubeletVersion, err := version.ParseSemantic(kubeletVersionStr)
+			kubeadmutil.CheckErr(err)
+
+			client, err := kubeconfigutil.ClientSetFromFile(*kubeConfigFile)
+			kubeadmutil.CheckErr(err)
+
+			// This call returns the ready-to-use configuration based on the configuration file
+			internalcfg, err := configutil.FetchConfigFromFileOrCluster(client, os.Stdout, "kubelet", cfgPath)
+			kubeadmutil.CheckErr(err)
+
+			err = kubeletphase.WriteKubeletConfigToDisk(internalcfg.KubeletConfiguration.BaseConfig, kubeletVersion)
 			kubeadmutil.CheckErr(err)
-			if features.Enabled(internalcfg.FeatureGates, features.DynamicKubeletConfig) {
-				client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
-				kubeadmutil.CheckErr(err)
-				err = kubeletphase.CreateBaseKubeletConfiguration(internalcfg, client)
-				kubeadmutil.CheckErr(err)
-			} else {
-				fmt.Println("[kubelet] feature gate DynamicKubeletConfig is not enabled, do nothing.")
-			}
 		},
 	}
 
+	cmd.Flags().StringVar(&kubeletVersionStr, "kubelet-version", kubeletVersionStr, "The desired version for the kubelet")
 	cmd.Flags().StringVar(&cfgPath, "config", cfgPath, "Path to kubeadm config file (WARNING: Usage of a configuration file is experimental)")
-	cmd.Flags().StringVar(&kubeConfigFile, "kubeconfig", "/etc/kubernetes/admin.conf", "The KubeConfig file to use when talking to the cluster")
-
 	return cmd
 }
 
 // NewCmdKubeletEnableDynamicConfig calls cobra.Command for enabling dynamic kubelet configuration on node
-func NewCmdKubeletEnableDynamicConfig() *cobra.Command {
-	cfg := &kubeadmapiv1alpha2.NodeConfiguration{}
-	kubeadmscheme.Scheme.Default(cfg)
+func NewCmdKubeletEnableDynamicConfig(kubeConfigFile *string) *cobra.Command {
+	var nodeName, kubeletVersionStr string
 
-	var cfgPath string
 	cmd := &cobra.Command{
-		Use:     "enable",
-		Aliases: []string{"update"},
-		Short:   "Enables or updates dynamic kubelet configuration on node",
+		Use:     "enable-dynamic-config",
```

Comment:
This is going to be off by default, right?  I don't see a message stating the default.

Reply:
yes, off by default. Now added a scary warning